crates/fuel-core/src/query/message.rs

```diff
@@ -135,30 +132,18 @@ impl<D: DatabasePort + ?Sized> MessageProofData for D {
     ) -> StorageResult<TransactionStatus> {
         self.status(transaction_id)
     }
-
-    fn transactions_on_block(&self, block_id: &BlockId) -> StorageResult<Vec<TxId>> {
-        self.block(block_id).map(|block| block.into_inner().1)
-    }
-
-    fn signature(&self, block_id: &BlockId) -> StorageResult<Signature> {
-        let consensus = self.consensus(block_id)?;
-        match consensus {
-            // TODO: https://github.com/FuelLabs/fuel-core/issues/816
-            Consensus::Genesis(_) => Ok(Default::default()),
-            Consensus::PoA(c) => Ok(c.signature),
-        }
-    }
 }
 
 /// Generate an output proof.
 // TODO: Do we want to return `Option` here?
 pub fn message_proof<T: MessageProofData + ?Sized>(
-    data: &T,
+    database: &T,
     transaction_id: Bytes32,
     message_id: MessageId,
```

Comment:
should we change this to nonce?

Reply:
`Nonce` or `MessageId` for us doesn't make any difference but may affect the end user. I left it `MessageId` because it is part of the verification process. Also, tests on our side and, I guess, on the contracts side, use `MessageId`=)